this is a diff, local/grub 2:2.02-8 and earlier
ensures the grub menu boot order is: linux-git then linux then linux-lts

--- orig/usr/share/grub/grub-mkconfig_lib	2017-12-30 10:27:17.752596835 +0100
+++ patched/usr/share/grub/grub-mkconfig_lib	2017-12-30 10:25:11.002419802 +0100
@@ -240,6 +240,24 @@ version_test_gt ()
 {
   version_test_gt_a="`echo "$1" | sed -e "s/[^-]*-//"`"
   version_test_gt_b="`echo "$2" | sed -e "s/[^-]*-//"`"
+  #echo "!$version_test_gt_a!$version_test_gt_b!" >&2
+  if test \( "$version_test_gt_a" == "linux-git" -a "$version_test_gt_b" == "linux-stable" \) -o \
+          \( "$version_test_gt_a" == "linux-stable" -a "$version_test_gt_b" == "linux-git" \); then
+    version_test_gt_a="$(cat /boot/${version_test_gt_a}-x86_64.kver | sed -e 's/\.r/-r/')" #file generated by PKGBUILD
+    version_test_gt_b="$(cat /boot/${version_test_gt_b}-x86_64.kver | sed -e 's/\.r/-r/')" #file generated by PKGBUILD
+    #note: for sort -V or sort -n to work properly versions should be like 5.1-r0.ge93c9c99a629-1-ARCHLINUX and 5.1.1.r0.gb724e9356404-1-ARCHLINUX instead of 5.1.1.r0.gb724e9356404-1-ARCHLINUX and 5.1.r0.ge93c9c99a629-1-ARCHLINUX  (shown in order of version: earlier to later)
+  else
+  if test "$version_test_gt_a" == "linux-git"; then
+    return 0 #yes it's greater, ie. linux-git vs linux-lts
+  elif test "$version_test_gt_b" == "linux-git"; then
+    return 1 #no it's not greater, ie. linux-lts vs linux-git
+  fi
+  if test "$version_test_gt_a" == "linux-lts" -a "$version_test_gt_b" == "linux"; then
+    return 1 # no, `linux-lts` isn't greater than `linux`  (we want `linux` before it!)
+  elif test "$version_test_gt_a" == "linux" -a "$version_test_gt_b" == "linux-lts"; then
+    return 0 # yes, `linux` is greater than `linux-lts` (same thing)
+  fi
+  fi #stable and git
   version_test_gt_cmp=gt
   if [ "x$version_test_gt_b" = "x" ] ; then
     return 0
#@@ -240,6 +240,17 @@ version_test_gt ()
# {
#   version_test_gt_a="`echo "$1" | sed -e "s/[^-]*-//"`"
#   version_test_gt_b="`echo "$2" | sed -e "s/[^-]*-//"`"
#+  #echo "!$version_test_gt_a!$version_test_gt_b!" >&2
#+  if test "$version_test_gt_a" == "linux-git"; then
#+    return 0 #yes it's greater, ie. linux-git vs linux-lts
#+  elif test "$version_test_gt_b" == "linux-git"; then
#+    return 1 #no it's not greater, ie. linux-lts vs linux-git
#+  fi
#+  if test "$version_test_gt_a" == "linux-lts" -a "$version_test_gt_b" == "linux"; then
#+    return 1 # no, `linux-lts` isn't greater than `linux`  (we want `linux` before it!)
#+  elif test "$version_test_gt_a" == "linux" -a "$version_test_gt_b" == "linux-lts"; then
#+    return 0 # yes, `linux` is greater than `linux-lts` (same thing)
#+  fi
#   version_test_gt_cmp=gt
#   if [ "x$version_test_gt_b" = "x" ] ; then
#     return 0
##@@ -240,6 +240,11 @@ version_test_gt ()
## {
##   version_test_gt_a="`echo "$1" | sed -e "s/[^-]*-//"`"
##   version_test_gt_b="`echo "$2" | sed -e "s/[^-]*-//"`"
##+  if test "$version_test_gt_a" == "linux-git"; then
##+    return 0 #yes it's greater, ie. linux-git vs linux-lts
##+  elif test "$version_test_gt_b" == "linux-git"; then
##+    return 1 #no it's not greater, ie. linux-lts vs linux-git
##+  fi
##   version_test_gt_cmp=gt
##   if [ "x$version_test_gt_b" = "x" ] ; then
##     return 0
